runtime/src/bank.rs

```diff
@@ -3630,6 +3638,30 @@ mod tests {
             bank.rent_collection_partitions(),
             vec![(14, 31, 32), (0, 0, 64), (0, 17, 64)]
         );
+        bank = Arc::new(new_from_parent(&bank));
+        assert_eq!(bank.rent_collection_partitions(), vec![(17, 18, 64)]);
+    }
+
+    #[test]
+    fn test_rent_eager_across_epoch_with_half_gap() {
+        let (genesis_config, _mint_keypair) = create_genesis_config(1);
+
+        let mut bank = Arc::new(Bank::new(&genesis_config));
+        assert_eq!(bank.rent_collection_partitions(), vec![(0, 0, 32)]);
+
+        bank = Arc::new(new_from_parent(&bank));
+        assert_eq!(bank.rent_collection_partitions(), vec![(0, 1, 32)]);
+        for _ in 2..15 {
+            bank = Arc::new(new_from_parent(&bank));
+        }
+        assert_eq!(bank.rent_collection_partitions(), vec![(13, 14, 32)]);
+        bank = Arc::new(Bank::new_from_parent(&bank, &Pubkey::default(), 32));
+        assert_eq!(
+            bank.rent_collection_partitions(),
+            vec![(14, 31, 32), (0, 0, 64)]
+        );
+        bank = Arc::new(new_from_parent(&bank));
+        assert_eq!(bank.rent_collection_partitions(), vec![(0, 1, 64)]);
```

Comment:
I know this is rather untasteful hard coding magic numbers. But, I want to first fix the bug and deploy the updated logic to tds along with #10206.